File: LeetCodeProblemsGo/LeetCode1513NumberOfSubstringsWithOnly1s/LeetCode1513NumberOfSubstringsWithOnly1s.go
package LeetCode1513NumberOfSubstringsWithOnly1s

func numSub(s string) int {
	var result int64 = 0
	var delta int64
	var mod int64 = 1e9 + 7
	consecutinveOnesCount := 0
	for _, c := range s {
		if c == '1' {
			consecutinveOnesCount++
			continue
		}
		if consecutinveOnesCount > 0 {
			delta = int64(consecutinveOnesCount)
			delta *= delta + 1
			delta /= 2
			delta %= mod
			result += delta
			result %= mod
			consecutinveOnesCount = 0
		}
	}
	if consecutinveOnesCount > 0 {
		delta = int64(consecutinveOnesCount)
		delta *= delta + 1
		delta /= 2
		delta %= mod
		result += delta
		result %= mod
		consecutinveOnesCount = 0
	}
	return int(result)
}

File: LeetCodeProblemsCMake/Problems/LeetCode1513NumberOfSubstringsWithOnly1s.cpp
//
// Created by saeed on 11/16/25.
//

#include "LeetCode1513NumberOfSubstringsWithOnly1s.h"
using namespace std;

int LeetCode1513NumberOfSubstringsWithOnly1s::numSub(string s) {
    long result{};
    long consecutiveOnesCount{};
    long mod=1e9+7;
    long delta{};
    for (auto c:s) {
        if (c=='1') {
            consecutiveOnesCount++;
            continue;
        }
        if (consecutiveOnesCount>0) {
            delta=consecutiveOnesCount;
            delta*=(consecutiveOnesCount+1);
            delta/=2;
            delta%=mod;
            result+=delta;
            result%=mod;
            consecutiveOnesCount=0;
        }

    }
    if (consecutiveOnesCount>0) {
        delta=consecutiveOnesCount;
        delta*=(consecutiveOnesCount+1);
        delta/=2;
        delta%=mod;
        result+=delta;
        result%=mod;
    }
    return result;
}
